### PR TITLE
[BUILD] Package release tarball

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

```
export RELEASE_VERSION=1.3.0-SNAPSHOT # keep same version with pom.xml
export SKIP_GPG=false                 # optional, default false
build/release/create-package.sh source
build/release/create-package.sh binary
```
```
➜  apache-kyuubi git:(asf-pub) ls -lh work/release
total 161256
-rw-r--r--  1 chengpan  staff    60M Aug  4 15:12 kyuubi-1.3.0-SNAPSHOT-bin.tgz
-rw-r--r--  1 chengpan  staff   488B Aug  4 15:12 kyuubi-1.3.0-SNAPSHOT-bin.tgz.asc
-rw-r--r--  1 chengpan  staff   212B Aug  4 15:12 kyuubi-1.3.0-SNAPSHOT-bin.tgz.sha512
-rw-r--r--  1 chengpan  staff    18M Aug  4 15:10 kyuubi-1.3.0-SNAPSHOT-source.tgz
-rw-r--r--  1 chengpan  staff   488B Aug  4 15:10 kyuubi-1.3.0-SNAPSHOT-source.tgz.asc
-rw-r--r--  1 chengpan  staff   215B Aug  4 15:10 kyuubi-1.3.0-SNAPSHOT-source.tgz.sha512
```

Planning to add release script and release doc in following PRs.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: build/dist [hunk 1]
```diff
@@ -145,20 +145,19 @@ HIVE_VERSION=$("$MVN" help:evaluate -Dexpression=hive.version $@ 2>/dev/null\
 
 echo "Building Kyuubi package of version $VERSION against Spark version - $SPARK_VERSION"
 
+SUFFIX="-$NAME"
 if [[ "$NAME" == "none" ]]; then
-  SPARK_HADOOP_VERSION_SUFFIX="-hadoop${SPARK_HADOOP_VERSION}"
-
   if [[ "$SPARK_PROVIDED" == "true" ]]; then
-    NAME="without-spark"
+    SUFFIX=""
```
Comment (pan3793) [inline build/dist:151]: suffix `-without-spark` has been removed.

File: build/dist [hunk 2]
```diff
@@ -217,22 +216,19 @@ if [[ "$SPARK_PROVIDED" != "true" ]]; then
...
 cp -r "$KYUUBI_HOME/docker" "$DISTDIR"
 
 if [[ "$MAKE_TGZ" == "true" ]]; then
-  TARDIR_NAME=kyuubi-$VERSION-bin-$NAME
+  TARDIR_NAME="kyuubi-$VERSION-bin$SUFFIX"
   TARDIR="$KYUUBI_HOME/$TARDIR_NAME"
   rm -rf "$TARDIR"
   cp -r "$DISTDIR" "$TARDIR"
-  tar czf "$TARDIR_NAME.tar.gz" -C "$KYUUBI_HOME" "$TARDIR_NAME"
+  tar czf "$TARDIR_NAME.tgz" -C "$KYUUBI_HOME" "$TARDIR_NAME"
```
Comment (pan3793) [inline build/dist:232]: changed `tar.gz` => `tgz`

File: NOTICE-binary [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+// Version 2.0, in this case for Apache Kyuubi
+// ------------------------------------------------------------------
+
+Apache Kyuubi
+Copyright 2021 and onwards The Apache Software Foundation.
+
+This product includes software developed at
+The Apache Software Foundation (https://www.apache.org/).
+
+=====================================================================
```
Comment (pan3793) [inline NOTICE-binary:12]: These text will be the first part in final `NOTICE-binary`

Comment (pan3793) [reply NOTICE-binary:12]: Run `build/release/create-package.sh binary` and unarchive tgz to check the final `NOTICE-binary`

**Comment by pan3793** [comment]
@ulysses-you Would you please have a check on failures of Minikube Integration Test? Seems not relative.

**Comment by ulysses-you** [comment]
@pan3793 yeah, it's not related. Not sure why this happened, I will do a follow up.

**Comment by yaooqinn** [comment]
kyuubi-incubating-1.3.0xxx?

**Comment by pan3793** [comment]
> kyuubi-incubating-1.3.0xxx?

Both `kyuubi-incubating-1.3.0` and `kyuubi-1.3.0-incubating` are acceptable, prefer the latter because no extra effort after graduating to TLP.

https://mvnrepository.com/artifact/org.apache.iceberg/iceberg-api

**Comment by yaooqinn** [comment]
kyuubi-1.3.0-incubating +1

**Comment by ulysses-you** [comment]
> @pan3793 yeah, it's not related. Not sure why this happened, I will do a follow up.

seems it's a random event .. I passed it with https://github.com/apache/incubator-kyuubi/pull/890/checks?check_run_id=3239593925

**Comment by pan3793** [comment]
OK, let me retry and see what happens.

File: docs/develop_tools/distribution.md [hunk 4]
```diff
@@ -29,12 +29,12 @@ For instance,
...
+This results a Kyuubi distribution named `kyuubi-{version}-bin-custom-name.tgz` for you.
 
 If you are planing to deploy Kyuubi where `spark` is provided, in other word, it's not required to bundle spark binary, use 
 
 ```bash
 ./build/dist --tgz --spark-provided
 ```
 
-Then you will get a Kyuubi distribution without spark binary named `kyuubi-{version}-bin-without-spark.tar.gz`.
+Then you will get a Kyuubi distribution without spark binary named `kyuubi-{version}-bin-without-spark.tgz`.
```
Comment (yaooqinn) [inline docs/develop_tools/distribution.md:40]: fix the name here？

File: docs/quick_start/quick_start.md [hunk 5]
```diff
@@ -33,7 +33,7 @@ Additionally, if you want to work with other Spark compatible systems or plugins
 To install Kyuubi, you need to unpack the tarball. For example,
 
 ```bash
-tar zxf kyuubi-1.0.2-bin-spark-3.0.1.tar.gz
+tar zxf kyuubi-1.0.2-bin-spark-3.0.1.tgz
```
Comment (yaooqinn) [inline docs/quick_start/quick_start.md:36]: how about using 1.3.0 here?

**Review by yaooqinn — APPROVED** [review]

**Review by yaooqinn — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merging for v1.3.0

**Pull request closed without merging**